compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -245,6 +249,12 @@ impl<'f> PerFunctionContext<'f> {
                     return true;
                 }
 
+                let allocation_aliases_parameter =
+                    aliases.any(|alias| self.make_array_references.contains(&alias));
+                if allocation_aliases_parameter == Some(true) {
```

Comment:
Why do we keep naming these `allocation_aliases_parameter` when (it looks like) only the first two check parameters?

The others mention function calls, make_arrays, terminator values, and aliased references. Seems inaccurate.

Reply:
> The others mention function calls, make_arrays, terminator values, and aliased references. Seems inaccurate.

Oops yes you are correct. This is an oversight. I will update the naming here.

Reply:
done